tests/jenkins/run_test_amzn_linux_gpu.sh

```diff
@@ -15,34 +17,34 @@ echo 'export JRE_HOME=${JAVA_HOME}/jre' >> ~/.profile
 echo 'export PATH=$PATH:/apache-maven-3.3.9/bin/:/usr/bin:${JAVA_HOME}/bin' >> ~/.profile
 source ~/.profile
 user=`id -u -n`
-make -j 4 || exit -1
+
+set -e
+
+make -j 4
 
 echo "BUILD cpp_test"
-make -j 4 test || exit -1
+make -j 4 test
 export MXNET_ENGINE_INFO=true
-for test in tests/cpp/*_test; do
-    ./$test || exit -1
-done
+./build/tests/cpp/mxnet_test
 
 echo "BUILD valgrind_test"
-for test in tests/cpp/*_test; do
-    valgrind ./$test || exit -1
-done
+valgrind ./build/tests/cpp/mxnet_test
+
 export MXNET_ENGINE_INFO=false
 export PYTHONPATH=${PWD}/python
 
 echo "BUILD python_test"
-nosetests --verbose tests/python/unittest || exit -1
-nosetests --verbose tests/python/train || exit -1
+nosetests --verbose tests/python/unittest
+nosetests --verbose tests/python/train
 
 echo "BUILD python3_test"
-nosetests3 --verbose tests/python/unittest || exit -1
-nosetests3 --verbose tests/python/train || exit -1
```

Comment:
I removed the || exit 1 because I put "set -e" at the top so that the script exits on any error